common/debug-interface/src/libra_trace.rs

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::json_log::JsonLogEntry;
+use once_cell::sync::Lazy;
+use std::time::Instant;
+
+pub const TRACE_EVENT: &str = "trace_event";
+pub const TRACE_EDGE: &str = "trace_edge";
+
+pub static TRACE_START: Lazy<Instant> = Lazy::new(Instant::now);
+
+#[macro_export]
+macro_rules! trace_event {
+    ($stage:expr, $node:tt) => {
+        trace_event!($stage; {$crate::format_node!($node), module_path!(), Option::<u64>::None})
```

Comment:
self calling looks scary, should we call these two different things?

Reply:
This type of self calling is fairly common pattern that I've seen in other rust crates
If you are worried about potential infinite call loop - this would be surfaced right away, since macro is evaluated during compile time, there is no danger of this suddenly breaking in production